src/widgets/capture/selectionwidget.cpp

```diff
@@ -215,9 +218,11 @@ void SelectionWidget::parentMouseMoveEvent(QMouseEvent* e)
     }
 
     auto geom = geometry();
-    float aspectRatio = (float)geom.width() / (float)geom.height();
     bool symmetryMod = qApp->keyboardModifiers() & Qt::ShiftModifier;
     bool preserveAspect = qApp->keyboardModifiers() & Qt::ControlModifier;
+    if(aspectRatio < 0 || ! preserveAspect){
```

Comment:
whats the reason for checking if the aspectRatio is < 0?

Reply:
I think because of line 194, this if statement means "if the user have not forced certain aspect ratio (and if it is not marked to be preserved), then ..."

But based on what I can see, aspect ratio will always be `-1.0` before this `if`. Unless in C++ the variables are not bound to the function scope and line 224 in changes the global value of the variable.